### ISSUE TITLE
Plannable Import: Support count

### ISSUE THREAD

**Issue by Hronom** [hashicorp/terraform#34227]
### Terraform Version

```shell
Terraform v1.5.4
on linux_amd64
```

### Use Cases

The ability to use `count` within an import block would unblock the use cases where the import block applied on resource that present only in one environment. For example: `prod` and not on `staging`.

In complex scenarios, where you are importing resources in a variety of environments, to resources that use `count`, the current way that the import block can be used is very limited. You would basically need to define separate import blocks for every resource in every environment, which becomes so complex that it verges on being totally unusable.

In our use case in this scenario we use the traditional terraform import command and script the import outside of the Terraform plan. This works, but adds 'pre' steps that need to be done before a plan can be safely applied in migration scenarios, which adds complexity, as well as involving completely custom code that brings its own risks.

A logical extension of the functionality of the import block would be to add support for `count`.

This would greatly reduce the complexity of complex migration scenarios.

As of now there no way to use it with resources that present only on one environment and not another.

### Attempted Solutions

Nothing works:
```
import {
  count               = var.environment == "prod" ? 1 : 0

  to = aws_cloudwatch_metric_alarm.alarm[0]
  id = "xxx"
}

resource "aws_cloudwatch_metric_alarm" "alarm" {
  count               = var.environment == "prod" ? 1 : 0
  alarm_name          = "xxx"
...
}
```
not work due luck of `count`

### References
- Discussion here https://discuss.hashicorp.com/t/conditional-terraform-import-blocks/55648
- https://github.com/hashicorp/terraform/issues/33624

**Comment by jbardin** [comment]
Duplicate of #33624
